pkgs/development/libraries/glib/setup-hook.sh

```diff
@@ -26,9 +26,9 @@ glibPreFixupPhase() {
 
 # gappsWrapperArgsHook expects GSETTINGS_SCHEMAS_PATH variable to be set by this.
 # Until we have dependency mechanism in generic builder, we need to use this ugly hack.
-if [[ " ${preFixupPhases:-} " =~ " gappsWrapperArgsHook " ]]; then
-    preFixupPhases+=" "
-    preFixupPhases="${preFixupPhases/ gappsWrapperArgsHook / glibPreFixupPhase gappsWrapperArgsHook }"
+if [[ " ${preFixupPhases[*]:-} " =~ " gappsWrapperArgsHook " ]]; then
+    # Insert glibPreFixupPhase before gappsWrapperArgsHook
+    replaceElement preFixupPhases gappsWrapperArgsHook glibPreFixupPhase gappsWrapperArgsHook
 else
-    preFixupPhases+=" glibPreFixupPhase"
+    appendToVar preFixupPhases glibPreFixupPhase
 fi
```

Comment:
> I wonder whether a function `appendToVarBefore var target value` would work better, which would insert right before `target` if that exists and otherwise appends to the end?
> 
> Would certainly remove some more repetition. But I can't estimate whether the code would be much more complex because of that.

The function implementation wouldn't change much, but the new function needs a good name. `appendToVarBefore` sounds self-contradicting. How about `insertToVarBefore` or `insertToVarBeforeOrAppend`?

Reply:
`insertToVarPreceding`?